snapshot/snapshot.go

```diff
@@ -564,9 +564,6 @@ func (o *snapshotter) cleanupSnapshotDirectory(ctx context.Context, dir string)
 	if err := o.fs.Unmount(ctx, mp); err != nil {
 		log.G(ctx).WithError(err).WithField("dir", mp).Debug("failed to unmount")
 	}
-	if err := os.RemoveAll(dir); err != nil {
-		return fmt.Errorf("failed to remove directory %q: %w", dir, err)
-	}
```

Comment:
Without the `os.RemoveAll` this function may need a name change. It is now more of a `unmountFilesystem`. But before that other than `cleanup` this function is also has call sites from `createSnapshot` [on error] and `Remove` which makes sense for the directory to be cleaned up. My question is if graceful termination of snapshotter should take a seperate call path or expand cleanup to have cleanup options.

Reply:
Your first point is correct, but the second point is hard to make a call on. I initially added to function headers but ultimately decided to make a new struct to store this information. It's a bit extra IMO, but also would make future changes easier to make (and really, I feel like a lot of our functions could be refactored to take data like this).

Could also add a `defaultCleanupConfig` if that proves useful in any form.

Reply:
When a boolean input modifies the behavior of the function, then to me that begins to indicate the function might be doing too much. Here the problem is amplified with two boolean inputs. It seems like we have a clear logical separation here in 1) unmount filesystem and 2) cleanup directory. Maybe this leans more into having two seperate implementations with explicit reponsibilities.

Reply:
Ah, ok, that makes sense to me. The previously passed in boolean (cleanupCommitted) was used to route to `getCleanupDirectories`, so I guess we'll keep that function name for now.

I had already split the function in two — unmounting, and cleaning, of which the latter calls the former. This is desirable behavior in most cases, including in `cleanup()`, which is called in both `o.Cleanup()` and `o.Close()`. Per the comment in the code, "Cleanup cleans up disk resources from removed or abandoned snapshots", so I created a new function instead of reusing the existing cleanup() function, which hopefully makes the separation of ideas more apparent.